### DISCUSSION TITLE
access to seacrch params

### DISCUSSION THREAD

**Discussion by abhanan369** [vercel/next.js#62391] [Help]
### Summary

There is no hook available in nextjs 12 version for accessing search params

**Comment by icyJoseph** [comment]
How do you want to access those? As part of a callback, or as trigger/part of the render pass?

**Comment by abhanan369** [comment]
As useSearchParams like in nextjs 14

**Reply by icyJoseph** [reply]
In Pages Router you can use `useRouter` from 'next/router`, and take `query` from there:

- https://nextjs.org/docs/pages/api-reference/functions/use-router#router-object

**Comment by abhanan369** [comment]
O right, Thanks man
